unittests/Parser/ParserTests.cpp

```diff
@@ -176,6 +176,24 @@ TEST(ParserTest, Errors) {
         "%1:i1 = extractvalue 42, 1 (hasSomethingElse\n"
         "infer %1\n",
         "<input>:2:1: expected hasExternalUses token" },
+      { "%0:i8 = var (range[2,90))\n",
+        "<input>:1:1: expected '=' for range as 'range='" },
+      { "%0:i32 = var (range=2,90))\n",
+        "<input>:1:1: expected '[' to specify lower bound of range" },
+      { "%0:i32 = var (range=[2:i32,90:i32))\n",
+        "<input>:1:1: expected lower bound of range" },
+      { "%0:i32 = var (range=[2,90:i32))\n",
+        "<input>:1:1: expected upper bound of range" },
+      { "%0:i32 = var (range=[2 90))\n",
+        "<input>:1:1: expected ',' after lower bound of range" },
+      { "%0:i32 = var (range=[2,90)\n",
+        "<input>:1:1: expected ')' to complete data flow fact string" },
```

Comment:
"data flow fact string" is awkward, how about
expected ')' to complete range

Reply:
this check is not just for range, it's outside the big switch statement for each dataflow fact (ref:Parser.cpp). This is in general for any kind of dfa.

Reply:
ok just remove "string" then, it doesn't make sense here